aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -130,6 +131,37 @@
    */
   public static final String S3FILEIO_ACL = "s3fileio.acl";
 
+  /**
+   * The implementation of LockManager to use for concurrent commits.
+   */
+  public static final String LOCK_IMPL = "lock.impl";
+
+  /**
+   * After the given time in milliseconds, the process will give up the attempt to acquire a lock.
+   * Default to 3 minutes.
+   */
+  public static final String LOCK_WAIT_MS = "lock.wait-ms";
+  public static final long LOCK_WAIT_MS_DEFAULT = TimeUnit.MINUTES.toMillis(3);
+
+  /**
+   * After the given time in milliseconds, the lock is automatically expired.
+   * Default to 20 minutes.
+   */
+  public static final String LOCK_EXPIRE_MS = "lock.expire-ms";
+  public static final long LOCK_EXPIRE_MS_DEFAULT = TimeUnit.MINUTES.toMillis(20);
+
+  /**
+   * The DynamoDB table used for locking.
+   * <p>
+   * The table name must be explicitly provided as an input when lock is enabled.
+   * This ensures the user is aware of the use of a DynamoDB table that might incur cost.
+   * Each glue catalog should use a different DynamoDB table for locking,
+   * otherwise tables with the same name from different catalog will block each other's commit.
+   * If the table with the given name does not exist,
+   * the table will be created at runtime with billing mode set to PAY_PER_REQUEST.
+   */
+  public static final String LOCK_DYNAMO_TABLE = "lock.dynamo.table";
```

Comment:
What about using `lock.table`? I'm not sure that "dynamo" needs to be in the property name because it doesn't provide much value.